### PR TITLE
Add "Wuhu Island" by tizona and Maxmos

### PR DESCRIPTION
**Pull request by TheRealPear**
### Map Name
Wuhu Island

### Gamemode(s)
King of the Hill

### Description & Team Sizes
This is the ported version of Wuhu Island from the original [TGM maps](https://github.com/WarzoneMC/tgm-maps) repo. The map is inspired by the Nintendo Wii Sport game series. It has 4 teams with minimum of 3 players each and maximum of 100 players.

### Screenshots
(will complete later)

### XML
https://github.com/TheRealPear/PublicMaps/blob/wuhu-island/KoTH/Standard/Wuhu%20Island/map.xml

### Map Download
https://github.com/TheRealPear/PublicMaps/tree/wuhu-island/KoTH/Standard/Wuhu%20Island

---

- [X] Contains a `map.xml` file with valid syntax and formatting.
- [X] The `map.xml` uses proto `1.4.0` or above (older or legacy maps are exempt, so long as the XML works).
- [X] The `map.xml` explicitly [sets the gamemode](https://pgm.dev/docs/modules/general/main#map-gamemode) (for stat-tracking).
- [X] The `map.xml` has correctly credited the map authors and contributors.
- [X] Contains `map.png` file (`290x246`).
- [X] The map was built for (or has been converted to) Minecraft 1.8.
- [X] Chunks have been pruned.
- [X] The map has been tested on a server running PGM.
- [X] I agree to submit this map under a commercial, [CC BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0/) license.

---

### Known Issues
~~- The spawn regions are placed in incorrect locations.~~
~~- The capture point's radius are off by one block, causing the hill to not change color on certain sides.~~

***IMPORTANT!!!***
- I am experimenting with giving all players Speed I effect and Feather Falling I enchantment given the large size of this map.

### CODE REVIEW

**Comment by TheRealPear** [comment]
Another thing I've forgot to list is that I've slightly modified the map to reflect the actual name of locations from Wii games as well as adding slabs for uphill paths for easier navigation and better lighting (with time cycle enabled).

File: KoTH/Standard/Wuhu Island/map.xml [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        <progress>
+            <region id="wuhu-town"/>
+        </progress>
+        <captured>
+            <region id="wuhu-town"/>
+        </captured>
+    </control-point>
+    <control-point name="The Candle" points="1"> <!-- Alternative name: "Lighthouse" -->
+        <capture>
+            <cylinder id="the-candle" base="-1372.5,32,476.5" radius="4" height="4"/>
```
Comment (Wesdial) [inline KoTH/Standard/Wuhu Island/map.xml:82]: `<cylinder id="the-candle" base="-1371.5,32,476.5" radius="4" height="4"/>`

File: KoTH/Standard/Wuhu Island/map.xml [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+        <progress>
+            <region id="the-candle"/>
+        </progress>
+        <captured>
+            <region id="the-candle"/>
+        </captured>
+    </control-point>
+	<control-point name="Cabana Lagoon" points="1"> <!-- Alternative name: "Club" -->
+        <capture>
+            <cylinder id="cabana-lagoon" base="-1295.5,31,321.5" radius="4" height="4"/>
```
Comment (Wesdial) [inline KoTH/Standard/Wuhu Island/map.xml:93]: `<cylinder id="cabana-lagoon" base="-1294.5,31,321.5" radius="4" height="4"/>`

File: KoTH/Standard/Wuhu Island/map.xml [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        <progress>
+            <region id="cabana-lagoon"/>
+        </progress>
+        <captured>
+            <region id="cabana-lagoon"/>
+        </captured>
+    </control-point>
+    <control-point name="Stillwater Grotto" points="1"> <!-- Alternative name: "Cave" -->
+        <capture>
+            <cylinder id="stillwater-grotto" base="-1278.5,25,388.5" radius="4" height="4"/>
```
Comment (Wesdial) [inline KoTH/Standard/Wuhu Island/map.xml:104]: `<cylinder id="stillwater-grotto" base="-1277.5,25,388.5" radius="4" height="4"/>`

File: KoTH/Standard/Wuhu Island/map.xml [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        <progress>
+            <region id="stillwater-grotto"/>
+        </progress>
+        <captured>
+            <region id="stillwater-grotto"/>
+        </captured>
+    </control-point>
+    <control-point name="Summer Castle" points="1"> <!-- Alternative name: "Castle" or "Summerstone Castle" -->
+        <capture>
+            <cylinder id="summer-castle" base="-1205.5,53,400.5" radius="4" height="4"/>
```
Comment (Wesdial) [inline KoTH/Standard/Wuhu Island/map.xml:115]: ` <cylinder id="summer-castle" base="-1204.5,53,400.5" radius="4" height="4"/>`

File: KoTH/Standard/Wuhu Island/map.xml [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        <progress>
+            <region id="summer-castle"/>
+        </progress>
+        <captured>
+            <region id="summer-castle"/>
+        </captured>
+    </control-point>
+    <control-point name="Mysterious Ruins" points="1"> <!-- Alternative name: "Ruins" -->
+        <capture>
+            <cylinder id="mysterious-ruins" base="-1146.5,30,269.5" radius="4" height="4"/>
```
Comment (Wesdial) [inline KoTH/Standard/Wuhu Island/map.xml:126]: `<cylinder id="mysterious-ruins" base="-1145.5,30,269.5" radius="4" height="4"/>`

File: KoTH/Standard/Wuhu Island/map.xml [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+        <progress>
+            <region id="mysterious-ruins"/>
+        </progress>
+        <captured>
+            <region id="mysterious-ruins"/>
+        </captured>
+    </control-point>
+    <control-point name="Maka Wuhu" points="1"> <!-- Alternative name: "Volcano" -->
+        <capture>
+            <cylinder id="maka-wuhu" base="-1129.5,63,369.5" radius="4" height="4"/>
```
Comment (Wesdial) [inline KoTH/Standard/Wuhu Island/map.xml:137]: `<cylinder id="maka-wuhu" base="-1128.5,63, 369.5" radius="4" height="4"/>`

File: KoTH/Standard/Wuhu Island/map.xml [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+        <progress>
+            <region id="maka-wuhu"/>
+        </progress>
+        <captured>
+            <region id="maka-wuhu"/>
+        </captured>
+    </control-point>
+    <control-point name="Cedar Tree Cave" points="1"> <!-- Alternative name: "Hill" or "Cedar Tree Tunnel" -->
+        <capture>
+            <cylinder id="cedar-tree-cave" base="-1120.5,45,422.5" radius="4" height="4"/>
```
Comment (Wesdial) [inline KoTH/Standard/Wuhu Island/map.xml:148]: ` <cylinder id="cedar-tree-cave" base="-1119.5,45,422.5" radius="4" height="4"/>`

File: KoTH/Standard/Wuhu Island/map.xml [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+        <progress>
+            <region id="cedar-tree-cave"/>
+        </progress>
+        <captured>
+            <region id="cedar-tree-cave"/>
+        </captured>
+    </control-point>
+    <control-point name="Cocoba Hotel" points="1"> <!-- Alternative name: "Resort" or "Tennis Court" -->
+        <capture>
+            <cylinder id="cocoba-hotel" base="-1036.5,21,437.5" radius="4" height="4"/>
```
Comment (Wesdial) [inline KoTH/Standard/Wuhu Island/map.xml:159]: ` <cylinder id="cocoba-hotel" base="-1035.5,21,437.5" radius="4" height="4"/>`

File: KoTH/Standard/Wuhu Island/map.xml [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+        <progress>
+            <region id="cocoba-hotel"/>
+        </progress>
+        <captured>
+            <region id="cocoba-hotel"/>
+        </captured>
+    </control-point>
+    <control-point name="19th Hole Hotel" points="1"> <!-- Alternative name: "Wedge" -->
+        <capture>
+            <cylinder id="nineteeth-hole-hotel" base="-1421.5,26,883.5" radius="4" height="4"/>
```
Comment (Wesdial) [inline KoTH/Standard/Wuhu Island/map.xml:170]: ` <cylinder id="nineteeth-hole-hotel" base="-1420.5,26,883.5" radius="4" height="4"/>`

File: KoTH/Standard/Wuhu Island/map.xml [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+    <team id="green-only">green</team>
+    <team id="blue-only">blue</team>
+    <team id="red-only">red</team>
+</filters>
+<!-- Regions -->
+<regions>
+    <!-- Definitions -->
+    <union id="spawn-protections">
+        <cuboid id="yellow-spawn-protection" min="-1150,30,369" max="-1158,36,377"/>
+        <cuboid id="green-spawn-protection" min="-1162,60,358" max="-1169,67,351"/>
```
Comment (Wesdial) [inline KoTH/Standard/Wuhu Island/map.xml:192]: `<cuboid id="green-spawn-protection" min="-1159.5,60,351.5" max="-1170.5,67,359.5"/>`

- Green's spawn region, appeared to be the only region that needed a fix, it was off by one block, and players could enter the spawn. This should fix it.

**Review by Wesdial — CHANGES_REQUESTED** [review]
- The spawn regions are placed in incorrect locations.

> Fixed, green's spawn region, was off by one block, allowing enemies to enter it.

- The capture point's radius are off by one block, causing the hill to not change color on certain sides.

> Fixed as well, Incorrect coordinates, off by a block. This should fix everything you have. (everything has been tested before proposing these changes. feel free to test them as well if you do so please.)

**Comment by TheRealPear** [comment]
All map issues should be resolved.

As for the Speed I effect and Feather Falling I enchantment, I will leave that up to the map developers on keeping or removing it from this map. I personally think it would benefit from a boost as it is quite a large map compared to what we have in this repo.

**Pull request merged**